lib/user.rb

```diff
@@ -0,0 +1,29 @@
+require 'pry'
+
+
+class User
+
+attr_reader :name, :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(who, what)
+    who.jokes << what
+  end
+
+  def joke_by_id(number)
+    # self.select {|number| self.jokes.id}
+      self.jokes.each do |search|
```

Comment:
i probably would name the block variable `joke` instead of `search`.  The block variable represents each individual element of the collection you're iterating over, which in this case, is an array of joke objects.  So each individual element is a single joke.